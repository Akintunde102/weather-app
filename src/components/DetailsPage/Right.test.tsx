import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import RightDetails from './Right';
import { MOCK_WEATHER_DETAILS } from '@/test/__mocks__/weather-details';

jest.mock('@/store/notes', () => ({
    __esModule: true,
    default: jest.fn(() => ({
        addNote: jest.fn(),
        notes: [],
        getNotesByLocationFullName: jest.fn()
    })),
}));

describe('RightDetails component', () => {
    test('renders add note section by default', () => {
        render(<RightDetails weatherDetails={MOCK_WEATHER_DETAILS} />);
        expect(screen.getByText('Add Note')).toBeInTheDocument();
    });

    test('renders list of notes section after adding a note', () => {


        render(<RightDetails weatherDetails={MOCK_WEATHER_DETAILS} />);

        const input = screen.getByPlaceholderText('Type.....');
        fireEvent.change(input, { target: { value: 'Test note' } });

        fireEvent.click(screen.getByText('Add Note'));

        fireEvent.click(screen.getByText('Add'));

        expect(screen.getByText('List of Notes')).toBeInTheDocument();
        expect(screen.queryByText('Add Note')).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        render(
            <RightDetails weatherDetails={MOCK_WEATHER_DETAILS} />
        );
        expect(screen.getByTestId("right-details")).toMatchSnapshot();
    });
});
