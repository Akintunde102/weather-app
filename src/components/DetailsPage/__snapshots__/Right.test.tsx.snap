// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightDetails component matches snapshot 1`] = `
<div
  class="rightContainer"
  data-testid="right-details"
>
  <div
    class="rightWeatherCards"
  >
    <div
      class="weatherCard"
      data-testid="weather-card"
      style="background: rgba(25, 25, 25, 0.7);"
    >
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="content"
          >
            <div
              class="left"
            >
              <img
                alt="icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <span>
                Wind
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="body"
      >
        <div
          class="dataItem"
        >
          <span>
            0
            <sup>
              Â°
            </sup>
          </span>
        </div>
        <div
          class="dataItemDetails"
        >
          <span>
            with a speed of 
            10
             m/h
          </span>
        </div>
      </div>
      <div
        class="footerItemDetails"
      >
        <span>
          Direction: 
          N
        </span>
      </div>
    </div>
    <div
      class="weatherCard"
      data-testid="weather-card"
      style="background: rgba(25, 25, 25, 0.7);"
    >
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="content"
          >
            <div
              class="left"
            >
              <img
                alt="icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <span>
                UV
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="body"
      >
        <div
          class="dataItem"
        >
          <span>
            0
          </span>
        </div>
        <div
          class="dataItemDetails"
        />
      </div>
      <div
        class="footerItemDetails"
      />
    </div>
    <div
      class="weatherCard"
      data-testid="weather-card"
      style="background: rgba(25, 25, 25, 0.7);"
    >
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="content"
          >
            <div
              class="left"
            >
              <img
                alt="icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <span>
                Visibility
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="body"
      >
        <div
          class="dataItem"
        >
          <span>
            10
             mi
          </span>
        </div>
        <div
          class="dataItemDetails"
        />
      </div>
      <div
        class="footerItemDetails"
      />
    </div>
    <div
      class="weatherCard"
      data-testid="weather-card"
      style="background: rgba(25, 25, 25, 0.7);"
    >
      <div
        class="header"
      >
        <div
          class="container"
        >
          <div
            class="content"
          >
            <div
              class="left"
            >
              <img
                alt="icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <span>
                Humidity
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="body"
      >
        <div
          class="dataItem"
        >
          <span>
            50
            %
          </span>
        </div>
        <div
          class="dataItemDetails"
        />
      </div>
      <div
        class="footerItemDetails"
      />
    </div>
  </div>
  <div
    class="notesContainer"
  >
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="content"
        >
          <div
            class="left"
          >
            <img
              alt="icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <span>
              Add Note
            </span>
          </div>
          <div
            class="right"
          >
            <button
              class="button"
            >
               
              Show List of Notes
            </button>
          </div>
        </div>
        <div
          class="divider"
          data-testid="divider"
        />
      </div>
      <div
        class="body"
      >
        <form
          class="addNoteFormContainer"
        >
          <textarea
            class="textArea"
            placeholder="Type....."
            required=""
          />
          <button
            class="button"
            type="button"
          >
            Add
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
